### ISSUE TITLE
RS0041 mis-fires on static property

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn-analyzers#3630]
#### Analyzer package

[Microsoft.CodeAnalysis.PublicApiAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers)

#### Package Version

v3.0.0

#### Diagnostic ID

Example: [RS0041](https://github.com/dotnet/roslyn-analyzers/blob/master/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md)

#### Repro steps

In a project with PublicAPI.Shipped.txt and PublicAPI.Unshipped.txt files and the `<Nullable>enable</Nullable>` property defined, add this source code:

```cs
    public static class Foo
    {
        public static Bar X { get; } = new Bar();

        public class Bar
        {
        }
    }
```

Add the API to the PublicAPI.txt file.

#### Expected behavior

No RS0041 warnings.

#### Actual behavior

RS0041 claims that "Symbol 'X.get' uses some oblivious reference types."

In my whole project, this mis-fire only happens when the property getter returns a nested type.

**Comment by sharwell** [comment]
Duplicate of #3596

**Comment by sharwell** [comment]
This is fixed for the 3.3.0 and newer releases of this package 👍
